src/lib.rs

```diff
@@ -302,15 +305,33 @@ impl Client {
 
     /// Starts a new INSERT statement.
     ///
+    /// # Validation
+    ///
+    /// If validation is enabled (default), `RowBinaryWithNamesAndTypes` input format is used.
+    /// When [`Client::insert`] method is called for this `table` for the first time,
+    /// it will fetch the table schema from the server, allowing to validate the serialized rows,
+    /// as well as write the names and types of the columns in the request header.
+    ///
+    /// Fetching the schema will happen only once per `table`,
+    /// as the schema is cached by the client internally.
+    ///
+    /// With disabled validation, the schema is not fetched,
+    /// and the rows serialized with `RowBinary` input format.
+    ///
     /// # Panics
+    ///
     /// If `T` has unnamed fields, e.g. tuples.
-    pub fn insert<T: Row>(&self, table: &str) -> Result<insert::Insert<T>> {
-        insert::Insert::new(self, table)
+    pub async fn insert<T: Row>(&self, table: &str) -> Result<insert::Insert<T>> {
+        if self.get_validation() {
+            let metadata = get_row_metadata::<T>(self, table).await?;
```

Comment:
This cache doesn't take into account a database, only the table name. We can use one provided to `with_database()` and forbid overriding it using `with_option()`